### ISSUE TITLE
WAFv2 acl rules should allow JSON body and oversize content handling options

### ISSUE THREAD

**Issue by MrHash** [hashicorp/terraform-provider-aws#25832]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The WAFv2 acl rule related resource should allow setting of the body field content-type as JSON and oversize body content handling options. Currently this does not seem to be possible via Terraform.

<img width="712" alt="Screen Shot 2022-07-15 at 13 45 06" src="https://user-images.githubusercontent.com/390925/179226294-7ea448a8-5eb4-4f39-b7ce-0cb6a8fd8a96.png">

Duplicates https://github.com/hashicorp/terraform-provider-aws/issues/25545 but keeping open as both issues have a large number of upvotes and contain useful comments.

### New or Affected Resource(s)

* aws_wafv2_web_acl
* aws_wafv2_rule_group

### Potential Terraform Configuration

```hcl
resource "aws_wafv2_web_acl" "example" {
  name  = "example"
  scope = "CLOUDFRONT"
  default_action {
    allow {}
  }
  rule {
    name     = "BlockXSS"
    priority = 10
    action {
      block {}
    }
    statement {
      xss_match_statement {
        field_to_match {
          body {}
        }
        text_transformation {
          priority = 1
          type     = "NONE"
        }
      }
    }
    visibility_config {
      metric_name                = "block-xss"
      cloudwatch_metrics_enabled = true
      sampled_requests_enabled   = true
    }
  }
  visibility_config {
    metric_name                = "all-rules"
    cloudwatch_metrics_enabled = true
    sampled_requests_enabled   = true
  }
}
```

### References

* https://docs.aws.amazon.com/waf/latest/developerguide/waf-rule-statement-fields-list.html#waf-rule-statement-request-component-json-body
* https://docs.aws.amazon.com/waf/latest/developerguide/waf-rule-statement-oversize-handling.html

**Comment by kyokoshome** [comment]
Is any update? AWS will blocked API call without Oversized OversizeHandling at Oct.1

**Comment by scottwestover** [comment]
I have submitted a pull request here https://github.com/hashicorp/terraform-provider-aws/pull/26506, to add support for the oversize handling for the existing body option.

**Comment by bschaatsbergen** [comment]
Please note that as per the docs [here](https://docs.aws.amazon.com/waf/latest/developerguide/waf-rule-statement-oversize-handling.html), Oversize Handling applies to to more places in the wafv2 web acl; [Body](https://docs.aws.amazon.com/waf/latest/APIReference/API_Body.html), [JsonBody](https://docs.aws.amazon.com/waf/latest/APIReference/API_FieldToMatch.html), [Headers](https://docs.aws.amazon.com/waf/latest/APIReference/API_Headers.html) and [Cookies](https://docs.aws.amazon.com/waf/latest/APIReference/API_Cookies.html).

**Comment by MrHash** [comment]
Resolved by #26506
